infoschema/infoschema.go

```diff
@@ -650,14 +667,42 @@ func (ts *SessionExtendedInfoSchema) SchemaByTable(tableInfo *model.TableInfo) (
 		return nil, false
 	}
 
-	if db, ok := ts.LocalTemporaryTables.SchemaByTable(tableInfo); ok {
-		return db, true
+	if ts.LocalTemporaryTables != nil {
+		if db, ok := ts.LocalTemporaryTables.SchemaByTable(tableInfo); ok {
+			return db, true
+		}
 	}
 
 	return ts.InfoSchema.SchemaByTable(tableInfo)
 }
 
+// UpdateTableInfo implements InfoSchema.SchemaByTable.
+func (ts *SessionExtendedInfoSchema) UpdateTableInfo(id int64, db *model.DBInfo, tableInfo table.Table) error {
+	if ts.MdlTables == nil {
+		ts.MdlTables = NewSessionTables()
+	}
+	ts.MdlTables.RemoveTable(db.Name, tableInfo.Meta().Name)
```

Comment:
Why should first remove exist one, then add one?

Reply:
Update it by remove && insert